app/src/main/java/ch/epfl/sdp/game/Server.java

```diff
@@ -76,27 +70,31 @@ public void start() {
 
     @Override
     public void update() {
-        if (counter <= 0) {
+
+        if(counter % (2 * GameThread.FPS) == 0) {
             sendUserPosition();
             sendGameArea();
             sendEnemies();
             sendItemBoxes();
             sendPlayersHealth();
             sendPlayersItems();
             checkPlayerStatus();
-            counter = 2 * GameThread.FPS + 1;
         }
 
-        --counter;
-
-        // update Players score every 10 seconds
-        if (scoreTimeCounter <= 0) {
+        // Update the current game score in 10 seconds
+        if (counter % (10 * GameThread.FPS) == 0 ) {
             Log.d(TAG, "update: update the score of player");
             updateInGameScoreOfPlayer();
-            scoreTimeCounter = 10 * GameThread.FPS + 1;
         }
-        --scoreTimeCounter;
 
+        if (counter % (30 * GameThread.FPS) == 0) {
```

Comment:
perhabs you can add a javadoc here

Reply:
For the code climate, it makes no sense why the code climate think they are duplicated, so we just ignore.